### PR TITLE
Automated MSI package building

### PR DESCRIPTION
**Pull request by mahor1221**

### CODE REVIEW

File: scripts/windows-bundler.sh [hunk 1]
```diff
@@ -73,14 +74,20 @@ cp "resources/Tachidesk Browser Launcher.bat" $release_name
 cp "resources/Tachidesk Debug Launcher.bat" $release_name
 cp "resources/Tachidesk Electron Launcher.bat" $release_name
 
-zip_name=$release_name.zip
-zip -9 -r $zip_name $release_name
```
Comment (AriaMoradi) [inline scripts/windows-bundler.sh]: keep the zip release, we could probably convert it into a standalone version some time?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -58,6 +58,9 @@ jobs:
           dependencies-cache-enabled: true
           configuration-cache-enabled: true
 
+      - name: Install wixl
+        run: sudo apt-get install -y wixl
```
Comment (AriaMoradi) [inline .github/workflows/publish.yml]: run this inside `windows-bundler.sh`

File: .github/workflows/build_push.yml [hunk 3]
```diff
@@ -67,6 +67,9 @@ jobs:
           echo "$genTag"
           echo "::set-output name=value::$genTag"
 
+      - name: Install wixl
+        run: sudo apt-get install -y wixl
```
Comment (AriaMoradi) [inline .github/workflows/build_push.yml]: run this inside `windows-bundler.sh`

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: scripts/Tachidesk-Server-x64.wxs [hunk 4]
```diff
@@ -0,0 +1,86 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+  <Product Id="*" UpgradeCode="*"
+    Version="0.6.0" Language="1033" Name="Tachidesk Server" Manufacturer="Suwayomi">
```
Comment (AriaMoradi) [inline scripts/Tachidesk-Server-x64.wxs]: do we have to manually update this with each release?

maybe do the windows 4 part version number here? `0.6.0.1033`

File: scripts/Tachidesk-Server-x64.wxs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        <Directory Id="INSTALLDIR" Name="Tachidesk-Server" >
+          <Directory Id="jre"/>
+          <Directory Id="electron"/>
+        </Directory>
+      </Directory>
+      <Directory Id="ProgramMenuFolder">
+        <Directory Id="ProgramMenuDir" Name="Tachidesk-Server">
+          <Component Id="ProgramMenuDir" Guid="*">
+            <RemoveFolder Id="ProgramMenuDir" On="uninstall"/>
+            <RegistryValue Root="HKCU" Key="Software\[Manufacturer]\[ProductName]" Type="string" Value="" KeyPath="yes"/>
```
Comment (AriaMoradi) [inline scripts/Tachidesk-Server-x64.wxs:24]: Is this correct? We don't use registry keys afaik?

File: scripts/windows-bundler.sh [hunk 6]
```diff
@@ -76,11 +76,25 @@ cp "resources/Tachidesk Electron Launcher.bat" $release_name
...
 
 # clean up from possible previous runs
 if [ -f ../server/build/$zip_name ]; then
   rm ../server/build/$zip_name
+if [ -f ../server/build/$msi_name ]; then
+  rm ../server/build/$msi_name
 fi
 
-mv $zip_name ../server/build/
+mv $zip_name $msi_name ../server/build/
```
Comment (AriaMoradi) [inline scripts/windows-bundler.sh]: this doesn't seem to be correct

File: scripts/windows-bundler.sh [hunk 7]
```diff
@@ -76,11 +76,25 @@ cp "resources/Tachidesk Electron Launcher.bat" $release_name
 zip_name=$release_name.zip
 zip -9 -r $zip_name $release_name
 
+# create msi package
+sudo apt-get install -y wixl
```
Comment (AriaMoradi) [inline scripts/windows-bundler.sh]: move this line inside the `if [ $CI = true ]; then` block

File: scripts/Tachidesk-Server-x64.wxs (hunk 5 above)
Comment (mahor1221) [reply scripts/Tachidesk-Server-x64.wxs:24]: It's necessary for a clean uninstall

docs: 
https://wixtoolset.org/documentation/manual/v3/howtos/files_and_registry/create_start_menu_shortcut.html
https://www.firegiant.com/wix/tutorial/getting-started/the-files-inside/

File: scripts/windows-bundler.sh (hunk 6 above)
Comment (mahor1221) [reply scripts/windows-bundler.sh]: Could you explain more. Shouldn't `$msi_name` be moved to `server/build` ?

Comment (AriaMoradi) [reply scripts/windows-bundler.sh]: mv takes two arguments, you are giving it 3

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: scripts/windows-bundler.sh (hunk 6 above)
Comment (mahor1221) [reply scripts/windows-bundler.sh]: But I can do this on my machine: `mv file1 file2 file3 destination`
Is it different on github's ubuntu?

Comment (mahor1221) [reply scripts/windows-bundler.sh]: I'll change it. It doesn't make any difference anyway

File: scripts/Tachidesk-Server-x64.wxs (hunk 4 above)
Comment (mahor1221) [reply scripts/Tachidesk-Server-x64.wxs]: I forgot this one. should be able to automate it

Comment (mahor1221) [reply scripts/Tachidesk-Server-x64.wxs]: 4 part versioning is not necessary. We could do it if you want

**Comment by mahor1221** [comment]
Should I close this PR and make another one to tidying commits up at the end ?

**Comment by Syer10** [comment]
Its just going to be squashed anyway, so its fine

**Comment by mahor1221** [comment]
Yeah It's mentioned in `CONTRIBUTING.md` too. I've just read it

File: scripts/windows-bundler.sh [hunk 8]
```diff
@@ -76,11 +77,25 @@ cp "resources/Tachidesk Electron Launcher.bat" $release_name
...
+else
+  wixl -D ProductVersion=$release_ver -D SourceDir=$release_name --arch x64 Tachidesk-Server-x64.wxs jre.wxs electron.wxs -o $msi_name
+fi
+
 rm -rf $release_name
 
 # clean up from possible previous runs
 if [ -f ../server/build/$zip_name ]; then
   rm ../server/build/$zip_name
+elif [ -f ../server/build/$msi_name ]; then
```
Comment (AriaMoradi) [inline scripts/windows-bundler.sh]: why `elif`? should be another separate if

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

**Pull request merged**